### ISSUE TITLE
Loss increases with training

### ISSUE THREAD

**Issue by sangeethkumar1997** [tensorflow/models#10047]
I am trying to train faster_rcnn_resnet101_v1_640x640_coco17_tpu-8 on custom data set.

The loss increases with each training step.

![Screenshot from 2021-06-05 23-48-23](https://user-images.githubusercontent.com/61541017/120901563-6db68800-c659-11eb-93fd-5aa236ba3059.png)
![Screenshot from 2021-06-05 23-49-26](https://user-images.githubusercontent.com/61541017/120901567-70b17880-c659-11eb-9b44-37e43ba12f5d.png)

Using the below config file.

`

model {
  faster_rcnn {
    num_classes: 1
    image_resizer {
      fixed_shape_resizer {
        height: 640
        width: 640
      }
    }
    feature_extractor {
      type: 'faster_rcnn_inception_resnet_v2_keras'
      batch_norm_trainable: true
    }
    first_stage_anchor_generator {
      grid_anchor_generator {
        scales: [0.25, 0.5, 1.0, 2.0]
        aspect_ratios: [0.5, 1.0, 2.0]
        height_stride: 16
        width_stride: 16
      }
    }
    first_stage_box_predictor_conv_hyperparams {
      op: CONV
      regularizer {
        l2_regularizer {
          weight: 0.0
        }
      }
      initializer {
        truncated_normal_initializer {
          stddev: 0.01
        }
      }
    }
    first_stage_nms_score_threshold: 0.0
    first_stage_nms_iou_threshold: 0.7
    first_stage_max_proposals: 300
    first_stage_localization_loss_weight: 2.0
    first_stage_objectness_loss_weight: 1.0
    initial_crop_size: 17
    maxpool_kernel_size: 1
    maxpool_stride: 1
    second_stage_box_predictor {
      mask_rcnn_box_predictor {
        use_dropout: false
        dropout_keep_probability: 1.0
        fc_hyperparams {
          op: FC
          regularizer {
            l2_regularizer {
              weight: 0.0
            }
          }
          initializer {
            variance_scaling_initializer {
              factor: 1.0
              uniform: true
              mode: FAN_AVG
            }
          }
        }
        share_box_across_classes: true
      }
    }
    second_stage_post_processing {
      batch_non_max_suppression {
        score_threshold: 0.0
        iou_threshold: 0.6
        max_detections_per_class: 100
        max_total_detections: 100
      }
      score_converter: SOFTMAX
    }
    second_stage_localization_loss_weight: 2.0
    second_stage_classification_loss_weight: 1.0
    use_static_shapes: true
    use_matmul_crop_and_resize: true
    clip_anchors_to_image: true
    use_static_balanced_label_sampler: true
    use_matmul_gather_in_matcher: true
  }
}

train_config: {
  batch_size: 4
  sync_replicas: true
  startup_delay_steps: 0
  replicas_to_aggregate: 8
  num_steps: 100000
  optimizer {
    momentum_optimizer: {
      learning_rate: {
        cosine_decay_learning_rate {
          learning_rate_base: .16
          total_steps: 100000
          warmup_learning_rate: 0
          warmup_steps: 2500
        }
      }
      momentum_optimizer_value: 0.9
    }
    use_moving_average: false
  }
  fine_tune_checkpoint_version: V2
  fine_tune_checkpoint: "/home/ubuntu/sangeeth/object_detection/faster_rcnn_inception_resnet_v2_640x640_coco17_tpu-8/checkpoint/ckpt-0"
  fine_tune_checkpoint_type: "detection"
  data_augmentation_options {
    random_horizontal_flip {
    }
  }
  data_augmentation_options {
    random_adjust_hue {
    }
  }

  data_augmentation_options {
    random_adjust_contrast {
    }
  }

  data_augmentation_options {
    random_adjust_saturation {
    }
  }

  data_augmentation_options {
     random_square_crop_by_scale {
      scale_min: 0.6
      scale_max: 1.3
    }
  }
  max_number_of_boxes: 100
  unpad_groundtruth_tensors: false
  use_bfloat16: true  # works only on TPUs
}
train_input_reader: {
  label_map_path: "/home/ubuntu/sangeeth/object_detection/label_map.txt"
  tf_record_input_reader {
    input_path: "/home/ubuntu/sangeeth/object_detection/smoke.record"
  }
}

eval_config: {
  metrics_set: "coco_detection_metrics"
  use_moving_averages: false
  batch_size: 1;
}

eval_input_reader: {
  label_map_path: "/home/ubuntu/sangeeth/object_detection/label_map.txt"
  shuffle: false
  num_epochs: 1
  tf_record_input_reader {
    input_path: "/home/ubuntu/sangeeth/object_detection/smoke.record"
  }
}
`

**Comment by akashAD98** [comment]
@sangeethkumar1997 may I know how complex your dataset is?? maybe that's a reason.

**Comment by sangeethkumar1997** [comment]
@akashAD98 I am trying to find identify the smoking in a image.
![Screenshot from 2021-06-11 12-05-19](https://user-images.githubusercontent.com/61541017/121641872-5fc79380-caad-11eb-8b00-3ed6f3dcb77c.png)
Without Data Augmentation the loss decreases in training dataset. But not well during performing well in prediction.

When using Data Augmentation, like scale [0.5-1.5] loss is not decreasing below certain point. This is the case with CenterNet HourGlass104 512x512 and  faster_rcnn_resnet101_v1_640x640_coco17_tpu-8

![Screenshot from 2021-06-11 12-09-48](https://user-images.githubusercontent.com/61541017/121642449-280d1b80-caae-11eb-855b-003a329b389b.png)
![Screenshot from 2021-06-11 12-13-47](https://user-images.githubusercontent.com/61541017/121642742-92be5700-caae-11eb-837b-79c812255819.png)

I am training on a dataset of 1000 images and one class.

Kindly let me know if there is any way to increase the performance.

**Comment by tucachmo2202** [comment]
@sangeethkumar1997 may you need more data. I think 1000 images are quite few.
